rules/coding-style/src/Rector/Variable/VariableVariablesRector.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\Variable;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\CodingStyle\Tests\Rector\Variable\VariableVariablesRector\VariableVariablesRectorTest
+ * @see https://www.php.net/manual/en/language.variables.variable.php
+ */
+final class VariableVariablesRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Ensure variable variables are wrapped in curly braces', [
+            new CodeSample(
+                <<<'PHP'
+function f()
+{
+    global $$foo->bar;
+}
+PHP
+                ,
+                <<<'PHP'
+function f()
+{
+    global ${$foo->bar};
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Variable::class];
+    }
+
+    /**
+     * @param Variable $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof Variable) {
+            return null;
+        }
+
+        $origNode = $node->getAttribute(AttributeKey::ORIGINAL_NODE);
+        if (! $origNode instanceof Variable) {
+            return null;
+        }
+
+        $parentNode = $node->getAttribute(AttributeKey::PARENT_NODE);
+        if ($parentNode instanceof Variable) {
+            if ($node->getStartTokenPos() !== $parentNode->getStartTokenPos()) {
+                return null;
+            }
+
+            if ($node->getEndTokenPos() !== $parentNode->getEndTokenPos()) {
+                return null;
+            }
+
+            if (! is_string($node->name)) {
+                return null;
+            }
+
+            $node->name = new Variable($node->name);
+
+            return $node;
```

Comment:
Could you extract this to own method? To separate this big long method complexity

Reply:
Turns out the code I had was more complicated than was needed! Simplified it to the point a separate method is not required